Tests fail on Python 3.13

A number of tests fail when typed-argument-parser is run on 3.13. Looks like some help output changed.

```
=========================== short test summary info ============================
FAILED tests/test_to_tap_class.py::test_subclasser_complex_help_message[function] - AssertionError: assert 'usage: pytes...sage and exit' == 'usage: pytes...sage and exit'
  
  Skipping 138 identical leading characters in diff, use -v to show
  - ions: -arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?                                          -----------------------------------------------------------------
  + ions: -arg, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?           ++++++++++++++++++++++++++++++++++
FAILED tests/test_to_tap_class.py::test_subclasser_complex_help_message[Class] - AssertionError: assert 'usage: pytes...sage and exit' == 'usage: pytes...sage and exit'
  
  Skipping 138 identical leading characters in diff, use -v to show
  - ions: -arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?                                          -----------------------------------------------------------------
  + ions: -arg, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?           ++++++++++++++++++++++++++++++++++
FAILED tests/test_to_tap_class.py::test_subclasser_complex_help_message[_Args] - AssertionError: assert 'usage: pytes...sage and exit' == 'usage: pytes...sage and exit'
  
  Skipping 138 identical leading characters in diff, use -v to show
  - ions: -arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?                                          -----------------------------------------------------------------
  + ions: -arg, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?           ++++++++++++++++++++++++++++++++++
FAILED tests/test_to_tap_class.py::test_subclasser_complex_help_message[class_or_function_3] - AssertionError: assert 'usage: pytes...sage and exit' == 'usage: pytes...sage and exit'
  
  Skipping 138 identical leading characters in diff, use -v to show
  - ions: -arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?                                          -----------------------------------------------------------------
  + ions: -arg, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME (Union[float, int], default=3) This argument has a long name and will be aliased with a short one --arg_int ARG_INT (int, required) some integer --arg_bool (bool, default=True) --arg_list [ARG_LIST ...] (Optional[List[str]], default=None) some list of strings -h, --help show this help message and exit
  ?           ++++++++++++++++++++++++++++++++++
======================== 4 failed, [299](https://github.com/python/typing_extensions/actions/runs/11285716477/job/31388972083?pr=484#step:10:300) passed in 1.32s =========================
```

Found this while working on python/typing_extensions#484.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -16,7 +16,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
 
     steps:
     - uses: actions/checkout@main

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,6 +29,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "License :: OSI Approved :: MIT License",
     "Operating System :: OS Independent",
     "Typing :: Typed",

--- a/tests/test_to_tap_class.py
+++ b/tests/test_to_tap_class.py
@@ -24,9 +24,14 @@ else:
     _IS_PYDANTIC_V1 = Version(pydantic.__version__) < Version("2.0.0")
 
 
-# To properly test the help message, we need to know how argparse formats it. It changed from 3.8 -> 3.9 -> 3.10
+# To properly test the help message, we need to know how argparse formats it. It changed from 3.8 -> 3.9 -> 3.10 -> 3.13
 _OPTIONS_TITLE = "options" if not sys.version_info < (3, 10) else "optional arguments"
 _ARG_LIST_DOTS = "..." if not sys.version_info < (3, 9) else "[ARG_LIST ...]"
+_ARG_WITH_ALIAS = (
+    "-arg, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME"
+    if not sys.version_info < (3, 13)
+    else "-arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME"
+)
 
 
 @dataclasses.dataclass
@@ -416,7 +421,7 @@ def test_subclasser_complex_help_message(class_or_function_: Any):
     {description}
 
     {_OPTIONS_TITLE}:
-    -arg ARGUMENT_WITH_REALLY_LONG_NAME, --argument_with_really_long_name ARGUMENT_WITH_REALLY_LONG_NAME
+    {_ARG_WITH_ALIAS}
                             (Union[float, int], default=3) This argument has a long name and will be aliased with a short
                             one
     --arg_int ARG_INT     (int, required) some integer

